src/path.ts

```diff
@@ -253,7 +253,10 @@ export const format: typeof path.format = function (p) {
 
 // basename
 export const basename: typeof path.basename = function (p, extension) {
-  const lastSegment = normalizeWindowsPath(p).split("/").pop();
+  const lastSegment = normalizeWindowsPath(p)
+    .replace(/\/+$/, "")
+    .split("/")
+    .pop();
```

Comment:
Perf: I think we can use two lines `const segments = []` and another check to see if last part is empty, fallback to len-1 this way we avoid regex

Reply:
> Perf: I think we can use two lines `const segments = []` and another check to see if last part is empty, fallback to len-1 this way we avoid regex

We need to ignore all trailing sep if there are many. See my test cases (`/foo/bar////`). If we loop and pop, it will be slower than regex.

Reply:
I think most of the fast and normal conditions is that there is no trailing slash or just one. We don't need to pop, we can just have a reverse for loop to find the first index that has non-empty one.

Feel free to do a benchmark. What matters to me is that we don't add a perf overhead for normal conditions by adding regex replace (since this is a low level utility, it could be used hundreds or thousands of times in high level tools)

Reply:
@pi0 Benchmark: https://replit.com/@isukkaw/SandyPossibleLanguage

TL; DR:

- `endsWith` before regex replace is fastest when the input doesn't end with sep. But it is slowest if there are any
- reverse loop and pop is fastest if there are seps at the end of the input. It is the second fastest if the input doesn't end with one (but it is still 10x slower than `endsWith` before regex replace)

Reply:
Thanks!! See this updated repl. I have  made few changes, removed other variants and add tests/adjustment that all implementations return basename for us. 

https://replit.com/@pi0/httpsgithubcomunjspathepull180

How do you think?

Reply:
@pi0 Just a friendly ping~